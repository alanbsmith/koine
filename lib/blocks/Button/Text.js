import styled from 'styled-components';
import { rem } from 'polished';

const Text = styled.span`
  font-size: ${rem(12)};
  font-weight: ${props => props.theme.fontWeights.light};
  text-transform: uppercase;
`;

export default Text;
