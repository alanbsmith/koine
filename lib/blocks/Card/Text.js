import styled from 'styled-components';
import { rem } from 'polished';

import P from '../../elements/P';

const Text = styled(P)`
  width: ${rem(240)};
`;

export default Text;
