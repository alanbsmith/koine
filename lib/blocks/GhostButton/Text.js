import styled from 'styled-components';
import gridScale from '../../utils/gridScale';

const Text = styled.span`
  font-size: ${props => gridScale(props, 1.5)};
  font-weight: ${props => props.theme.fontWeights.light};
  text-transform: uppercase;
`;

export default Text;
