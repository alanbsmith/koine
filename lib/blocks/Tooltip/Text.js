import styled from 'styled-components';
import { applyStyleModifiers } from 'styled-components-modifiers';

import { fontStyles, fontWeights } from '../../modifiers';
import gridScale from '../../utils/gridScale';

const modifierConfig = {
  ...fontStyles,
  ...fontWeights,
};

const Text = styled.span`
  border-radius: ${props => props.theme.dimensions.borderRadius};
  color: ${props => props.theme.colors.base.chrome000};
  font-size: ${props => gridScale(props, 1.5)};
  font-weight: ${props => props.theme.fontWeights.light};
  text-transform: uppercase;
  ${applyStyleModifiers(modifierConfig)}
`;

export default Text;
