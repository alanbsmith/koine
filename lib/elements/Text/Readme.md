```jsx
<div>
  <Text modifiers={['capitalize']}>hello world</Text>
</div>
```
