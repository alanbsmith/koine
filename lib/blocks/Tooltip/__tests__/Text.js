import React from 'react';
import Text from '../Text';
import 'jest-styled-components';

import { renderWithTheme } from '__tests__/helpers';


describe('Tooltip Text Element', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(
      <Text>
        Tooltip Text
      </Text>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('applies the capitalize modifier', () => {
    const tree = renderWithTheme(
      <Text modifiers={['capitalize']}>
        Tooltip Text
      </Text>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('applies the boldWeight modifier', () => {
    const tree = renderWithTheme(
      <Text modifiers={['boldWeight']}>
        Tooltip Text
      </Text>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
