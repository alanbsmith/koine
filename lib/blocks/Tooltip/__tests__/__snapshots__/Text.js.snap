// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Text Element applies the boldWeight modifier 1`] = `
.c0 {
  border-radius: 2px;
  color: #FFFFFF;
  font-size: 0.75rem;
  font-weight: 300;
  text-transform: uppercase;
  font-weight: 700;
}

<span
  className="c0"
>
  Tooltip Text
</span>
`;

exports[`Tooltip Text Element applies the capitalize modifier 1`] = `
.c0 {
  border-radius: 2px;
  color: #FFFFFF;
  font-size: 0.75rem;
  font-weight: 300;
  text-transform: uppercase;
  text-transform: capitalize;
}

<span
  className="c0"
>
  Tooltip Text
</span>
`;

exports[`Tooltip Text Element renders correctly 1`] = `
.c0 {
  border-radius: 2px;
  color: #FFFFFF;
  font-size: 0.75rem;
  font-weight: 300;
  text-transform: uppercase;
}

<span
  className="c0"
>
  Tooltip Text
</span>
`;
